### ISSUE TITLE
`InvariantError` with `ApolloProvider` when client should be set

### ISSUE THREAD

**Issue by callensm** [apollographql/apollo-client#10938]
**Environment**: Browser
**Version**: `>=3.8.0-alpha.14`

Using `@apollo/client` versioning `3.8.0-alpha.14`+, I'm receiving this invariant error from the `ApolloProvider` context component claiming that the client is not available for use within my query hooks. I know this to not be true because the application does not even mount unless the client is set (see snippet below).

This error persists through to the current `3.8.0-beta.0` version as well.

## Snippets

```tsx
export function App() {
  const user = useAuthenticatedUser();
  if (!user) {
    return <Loading />;
  }

  const apolloClient = createApolloClient(user.jwt);

  return (
    <ApolloProvider client={apolloClient} suspenseCache={suspenseCache}>
          <div
            style={{
              display: "flex",
              flexDirection: "column",
              height: "100%",
            }}
          >
            <Router />
          </div>
    </ApolloProvider>
  );
}
```

## Screenshots

![](https://cdn.discordapp.com/attachments/1113323302375653497/1113521439073652786/image.png)

**Comment by phryneas** [comment]
We changed the way the Context object is created/deduplicated around that time. That should not make any problems normally, but I can imagine one scenario where you might run into problems: it you have the `@apollo/client` package installed in one before-client and one after-client version side by side. Could you check with `npm ls @apollo/client`/`yarn why @apollo/client` if that could be the case?

**Comment by alessbell** [comment]
Hi @callensm 👋 I'm trying to isolate the potential source of this issue, would you be able to try this snapshot release: `npm i @apollo/client@0.0.0-pr-10954-20230606145830` and let me know if it solves your problem? A fix should be out in one of the next betas once we've confirmed the issue, thanks!

**Comment by jerelmiller** [comment]
Confirmed the snapshot build works via Discord: https://discord.com/channels/1022972389463687228/1113323302375653497/1116755935617106041

**Comment by jerelmiller** [comment]
Fixed with #10957
